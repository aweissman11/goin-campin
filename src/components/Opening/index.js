import React, { Component } from 'react';

export class Opening extends Component {
  render() {
    return (
      <div className='intro-container-container'>
        <div 
          className='intro-animation-container'
          onClick={() => this.props.completeOpening()}
        >
        </div>
      </div>
    )
  }
}

export default Opening;