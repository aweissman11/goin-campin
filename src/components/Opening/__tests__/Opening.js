import React from 'react';
import { shallow } from 'enzyme';

import Opening from '../index';

describe('Opening', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Opening completeOpening={jest.fn()} />)
    expect(wrapper).toMatchSnapshot()
  })
  
  it('should fire completeOpening onClick', () => {
    const mockComplete = jest.fn();
    const wrapper = shallow(<Opening completeOpening={mockComplete} />)
    wrapper.find('.intro-animation-container').simulate('click');
    expect(mockComplete).toHaveBeenCalled()
  })
  
  
})
