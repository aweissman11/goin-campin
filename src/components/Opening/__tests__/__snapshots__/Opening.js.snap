// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Opening should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Opening
    completeOpening={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="intro-animation-container"
        onClick={[Function]}
      />,
      "className": "intro-container-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "intro-animation-container",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="intro-animation-container"
          onClick={[Function]}
        />,
        "className": "intro-container-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "intro-animation-container",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
